src/nac/management/commands/import_devices_from_csv.py

```diff
@@ -93,35 +113,66 @@ def handle_deviceObject(self, deviceObject):
             self.save_invalid_devices(deviceObject)
         except Exception as e:
             logging.error(f"Error: Handling device Object failed -> {e}")
+            traceback.print_exc()
+
+    def get_deviceRole(self, deviceObject):
```

Comment:
This function is quite long. You should consider to move all the stuff inside the if and the else parts into separate functions. this would improve readability.